src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -115,11 +115,17 @@ public static double Abs(Complex value)
                 return double.PositiveInfinity;
             }
 
+            return Hypot(value._real, value._imaginary);
+        }
+
+        private static double Hypot (double a, double b)
+        {
             // |value| == sqrt(a^2 + b^2)
```

Comment:
There are actually two issues here. Although both go back to inf/inf divisions, they are inf/inf divisions in different places; we can't fix both by one change.

One issue is that Hypot doesn't deal with (inf,inf) correctly: it returns NaN rather than +inf, because internally it does an inf/inf division. This only happens if both arguments are infinite; if one is finite, it correctly returns +inf.

The other is that, even if Hypot is fixed, Sqrt(inf, inf) will still return NaN (for one of the components, not both), because it will be doing another inf/inf division after it gets inf back from Hypot. Again, this only happens in both components are infinite; if one is finite, it will correctly return with one component infinite, with the right sign, and the other one zero.

The actual behavior we want for Sqrt for double infinities is determined by which quadrants square root maps into which, namely:
(+inf,+inf) -> (+inf,+inf)
(+inf,-inf) -> (+inf,-inf)
(-inf, -inf) -> (+inf, -inf)
(-int,+inf) -> (+inf,-inf)
It's probably best put this logic on the "slow path" inside the rescaling clause, because then we aren't adding any additional tests inside the "fast path".